table/tables/partition.go

```diff
@@ -313,7 +313,13 @@ func (t *partitionedTable) locateHashPartition(ctx sessionctx.Context, pi *model
 
 // GetPartition returns a Table, which is actually a partition.
 func (t *partitionedTable) GetPartition(pid int64) table.PhysicalTable {
-	return t.partitions[pid]
+	// Attention, can't simply use `return t.partitions[pid]` here.
+	// Because A nil of type *partition is a kind of `table.PhysicalTable`
```

Comment:
```suggestion
	// Because A nil of type *partition is a kind of `table.PhysicalTable`,
	// the final result will be a non-nil interface whose underlying value is nil,
	// use that result will get a nil pointer panic.
```